ParkingDatabase/src/main/java/com/example/parkingdatabase/ListCarsUsingParking.java

```diff
@@ -0,0 +1,37 @@
+package com.example.parkingdatabase;
+
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.ListIterator;
+
+public class ListCarsUsingParking {
+
+    private final ParkingService parkingService;
+
+    public ListCarsUsingParking(ParkingService parkingService) {
```

Comment:
I think that your problem is that you don't inject any reference to ParkingService here.
I would suggest to remove this class and move this logic to ParkingService. You have @Autowired CarRepo there, so you can just use it